tests/e2e/utils/pipelines/setup_pipelines_irsa.py

```diff
@@ -0,0 +1,112 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import json
+import boto3
+
+from e2e.utils.utils import (
+    load_json_file,
+    get_iam_client,
+    get_eks_client,
+)
+from e2e.fixtures.cluster import (
+    associate_iam_oidc_provider,
+)
+from e2e.utils.aws.iam import IAMPolicy
+from e2e.utils.pipelines import common
+from e2e.utils.config import configure_env_file
+from e2e.utils.utils import print_banner, load_yaml_file, write_env_to_yaml, exec_shell
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def profile_trust_policy(cluster, region, account_id):
+    eks_client = get_eks_client(region=region)
+
+    resp = eks_client.describe_cluster(name=cluster)
+    oidc_url = resp["cluster"]["identity"]["oidc"]["issuer"].split("https://")[1]
+
+    trust_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        f"{oidc_url}:aud": "sts.amazonaws.com",
+                        f"{oidc_url}:sub": [
+                            "system:serviceaccount:kubeflow:ml-pipeline",
+                        ],
+                    }
+                },
+            }
+        ],
+    }
+    return json.dumps(trust_policy)
+
+
+def create_pipeline_oidc_role(cluster_name, region):
+    iam_client = get_iam_client(region=region)
+    acc_id = get_account_id()
+    role_name = f"{common.PIPELINE_OIDC_ROLE_NAME_PREFIX}-{cluster_name}"
```

Comment:
Can you limit the length of this string to 64 characters, IAM roles can have 64 [characters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-rolename) but an eks cluster name can have upto 100 [characters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-cluster.html#cfn-eks-cluster-name)